system/Cache/Handlers/BaseHandler.php

```diff
@@ -14,12 +14,65 @@
 use Closure;
 use CodeIgniter\Cache\CacheInterface;
 use Exception;
+use InvalidArgumentException;
 
 /**
  * Base class for cache handling
  */
 abstract class BaseHandler implements CacheInterface
 {
+	/**
+	 * Reserved characters that cannot be used in a key or tag.
+	 *
+	 * @see https://github.com/symfony/cache-contracts/blob/c0446463729b89dd4fa62e9aeecc80287323615d/ItemInterface.php#L43
+	 */
+	public const RESERVED_CHARACTERS = '{}()/\@:';
+
+	/**
+	 * Maximum key length.
+	 */
+	public const MAX_KEY_LENGTH = PHP_INT_MAX;
+
+	/**
+	 * Prefix to apply to cache keys.
+	 * May not be used by all handlers.
+	 *
+	 * @var string
+	 */
+	protected $prefix;
+
+	/**
+	 * Validates a cache key according to PSR-6.
+	 * Keys that exceed MAX_KEY_LENGTH are hashed.
+	 *
+	 * @param string $key    The key to validate
+	 * @param string $prefix Optional prefix to include in length calculations
+	 *
+	 * @throws InvalidArgumentException When $key is not valid
+	 *
+	 * @see https://github.com/symfony/cache/blob/7b024c6726af21fd4984ac8d1eae2b9f3d90de88/CacheItem.php#L158
+	 */
+	public static function validateKey($key, $prefix = ''): string
```

Comment:
If all of the handlers are based on this class, why make this static? It's already being instantiated so there's no savings there.

Reply:
Because this will be used by the PSR-6 adapter, which was the inspiration for this to begin with. I originally had this as a static method on the Cache Item class but I thought it would be a benefit to the framework Cache core - then I realized we had this bug with `FIleHandler` and long keys and it made sense to kill two issues with one static method.

Reply:
I thought it might have had to do with PSR6 so I checked that last night but didn't see where the interface required it be static? Or is it just that you need it usable by the other class? If that's the case, I'm cool.

Reply:
It's not that it is required to be static, but rather that the adapter will want to handle verification as well. We could have this be a class protected method and let the PSR-6 adapter have it own but it would be almost entirely duplicate code. Making this static here lets us re-use it outside of a specific handler's context.